contracts/security-token/RestrictedTransferAgent.sol

```diff
@@ -24,7 +23,16 @@ contract RestrictedTransferAgent is SecurityTransferAgent {
       return value;
     }
 
-    if (KYC.isWhitelisted(to)) {
+    // The following ifs might seem counterintuitive, but this function will
+    // be executed upon every transfer, so we want to make sure that despite of
+    // Solidity's optimization settings, only relevant external calls are made
+    if (KYC.getFlag(to, 0)) {
```

Comment:
The kyc system handles 256 types. We need better abstraction here to handle all kinds of flags. I don't know how though as I don't know all the possible use cases.

Reply:
Yes: we don't know that neither :slightly_smiling_face: Hence the responsibility of future flags is lifted to the future code (meaning, that more there are flags, better organization methods are needed to manage those in caller's side, it's also worth of noting that caller can be our backend or another smart contract, such as transfer agent, payout contract, or voting contract)